From af25cd7a2f4b14bf3efb8c5ffe8c7f0a3785426a Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Tue, 7 Sep 2021 14:25:44 +0300
Subject: [PATCH] Fix build issue with conflicting declarations in GL headers

Some graphics stacks have old version of GL/GLES headers which have conflicting
declaration and fail when GL and GLES headers are used at the same time.

/usr/include/GL/glext.h:466:19: error: conflicting declaration 'typedef ptrdiff_t GLsizeiptr'
 | typedef ptrdiff_t GLsizeiptr;
 |                   ^~~~~~~~

/usr/include/GLES3/gl3.h:75:25: note: previous declaration as 'typedef khronos_ssize_t GLsizeiptr'
 | typedef khronos_ssize_t GLsizeiptr;
 |                         ^~~~~~~~~~

Pick-to: 6.2
Change-Id: I98b96755ea1b27c942214adf3d76e65539302053
---
 src/multimedia/platform/gstreamer/common/qgstvideobuffer.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/multimedia/platform/gstreamer/common/qgstvideobuffer.cpp b/src/multimedia/platform/gstreamer/common/qgstvideobuffer.cpp
index 6511855c1..7cd022a82 100644
--- a/src/multimedia/platform/gstreamer/common/qgstvideobuffer.cpp
+++ b/src/multimedia/platform/gstreamer/common/qgstvideobuffer.cpp
@@ -54,6 +54,7 @@
 #include <QtGui/private/qrhigles2_p.h>
 #include <QtGui/qopenglcontext.h>
 #include <QtGui/qopenglfunctions.h>
+#include <QtGui/qopengl.h>
 
 #include <gst/gl/gstglconfig.h>
 #include <gst/gl/gstglmemory.h>
@@ -62,8 +63,6 @@
 
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
-#define GL_GLEXT_PROTOTYPES
-#include <GL/gl.h>
 #endif
 
 QT_BEGIN_NAMESPACE
